libs/coin-framework/src/api/types.ts

```diff
@@ -5,18 +5,28 @@ export type BlockInfo = {
 };
 
 export type Operation = {
-  hash: string;
-  address: string;
   type: string;
-  value: bigint;
-  fee: bigint;
-  block: BlockInfo;
+  // This operation corresponds to the index-th event triggered bu the original transaction
+  operationIndex: number;
   senders: string[];
   recipients: string[];
-  date: Date;
-  transactionSequenceNumber: number;
+  value: bigint;
+  // Asset is not defined when dealing with native currency
+  asset?: AssetInfo;
   // Field containing dedicated value for each blockchain
   details?: Record<string, unknown>;
+  tx: {
+    // One tx can trigger multiple operations, hence multiple operations with the same hash
+    hash: string;
+    // In which block this operation's related tx was included
+    block: BlockInfo;
+    fees: bigint;
+  };
+};
+
+export type AssetInfo = {
+  standard: string;
+  tokenAddressOrId: string;
```

Comment:
I'm afraid this trick will not work everywhere:
 - in some cases it might not be possible to distinguish if this a token id or contract address, and we will need to (I have not concrete example though)
 - on Algorand you need both contract address and token Id: 1 contract can deploy several tokens (for examples, look at first lines from https://github.com/LedgerHQ/crypto-assets/blob/main/inputs/tokens.csv)
 
 I suggest just aligning with CAL to make everyone's life easier:
```suggestion
  contractAddress?: string;
  token_identifier?: string;
```

Reply:
I agree with you, i'm more for splitting both, looks weird to me to pass both kinds of value in the same variable

Reply:
Discussed on slack, generic type